pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+        JoinedValidatorPool(T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        /// Round, Validator Account, Total Exposed Amount (includes all nominations)
+        ValidatorChosen(SessionIndex, T::AccountId, BalanceOf<T>),
+        /// Validator Account, Old Bond, New Bond
+        ValidatorBondedMore(T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        /// Validator Account, Old Bond, New Bond
+        ValidatorBondedLess(T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        ValidatorDeActive(SessionIndex, T::AccountId),
+        ValidatorActive(SessionIndex, T::AccountId),
+        /// Round, Validator Account, Scheduled Exit
+        ValidatorScheduledExit(SessionIndex, T::AccountId, SessionIndex),
+        /// Account, Amount Unlocked, New Total Amt Locked
+        ValidatorLeft(T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        // Nominator, Validator, Old Nomination, New Nomination
+        NominationIncreased(T::AccountId, T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        // Nominator, Validator, Old Nomination, New Nomination
+        NominationDecreased(T::AccountId, T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        /// Nominator, Amount Unstaked
+        NominatorLeft(T::AccountId, BalanceOf<T>),
+        /// Nominator, Amount Locked, Validator, New Total Amt backing Validator
+        Nomination(T::AccountId, BalanceOf<T>, T::AccountId, BalanceOf<T>),
+        /// Nominator, Validator, Amount Unstaked, New Total Amt Staked for Validator
+        NominatorLeftValidator(T::AccountId, T::AccountId, BalanceOf<T>, BalanceOf<T>),
+        /// Paid the account (nominator or validator) the balance as liquid rewards
+        Rewarded(T::AccountId, BalanceOf<T>),
+        /// Round inflation range set with the provided annual inflation range
+        RoundInflationSet(Perbill, Perbill, Perbill),
+        /// Staking expectations set
+        StakeExpectationsSet(BalanceOf<T>, BalanceOf<T>, BalanceOf<T>),
+        /// Set total selected validators to this value [old, new]
+        TotalSelectedSet(u32, u32),
+        /// Set validator commission to this value [old, new]
+        ValidatorCommissionSet(Perbill, Perbill),
+        /// Set blocks per round [current_round, first_block, old, new]
+        BlocksPerRoundSet(SessionIndex, T::BlockNumber, u32, u32),
+        /// One validator (and its nominators) has been slashed by the given amount.
+        /// \[validator, amount\]
+        Slash(T::AccountId, BalanceOf<T>),
+        /// An old slashing report was discarded because it could
+        /// not be processed. \[session_index\]
+        OldSlashingReportDiscarded(SessionIndex),
+        PayReporterReward(T::AccountId, BalanceOf<T>),
+        DeferredUnappliedSlash(SessionIndex, T::AccountId),
+        /// An account has called `withdraw_unbonded` and removed unbonding chunks worth `Balance`
+        /// from the unlocking queue. \[controller_account, amount\]
+        Withdrawn(T::AccountId, BalanceOf<T>),
```

Comment:
Let's make sure all events are documented. In addition to that we need all of them to have a parameter description in the form of `\[example1, example2\]` for the upcoming block explorer

Reply:
:+1:  Done